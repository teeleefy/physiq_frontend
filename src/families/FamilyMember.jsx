import {
    Card,
    CardBody,
    CardText,
    ListGroupItem
  } from "reactstrap";
  import { NavLink } from "react-router-dom";
  import './styles/FamilyMember.css'


function FamilyMember({familyMember}){

    return(
        <>
        <ListGroupItem>
            <Card className="FamilyMember-card">
                <CardBody>
                    <CardText>
                        <p><b>Family Member Name: </b>  
                            <NavLink className="FamilyMember-name" to={`/member/${familyMember.id}`} key={familyMember.id}>
                                 {familyMember.firstName} {familyMember.lastName}
                            </NavLink>
                        </p>
                        <p><b>Birthday:</b> {familyMember.birthday}</p>
                    </CardText>
                </CardBody>
            </Card>
        </ListGroupItem>    
        </>
     )
}

export default FamilyMember;