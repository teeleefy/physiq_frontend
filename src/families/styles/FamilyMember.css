#FamilyMember-main {
    width: 90vw;
    height: 82vh;
    background-color: white;
    overflow: scroll;
    color: black;
    border: 15px red solid;
    border-radius: 15px;
    padding: 0;
    margin: 0;
  }


.FamilyMember-card {
    height: 15vh;
    margin-left: 0.5rem;
    margin-bottom: 0.5rem;
}

.FamilyMember-label {
    font-size: 1.5rem;
}

.FamilyMember-h1 {
    font-size: 3rem;
    font-weight: 600;
}

.FamilyMember-date-input{
    width: 20vw;
}