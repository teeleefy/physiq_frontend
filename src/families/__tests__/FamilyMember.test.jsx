import { renderWithRouter, screen } from '../../../__tests__/test-tools';
import FamilyMember from '../FamilyMember'

let familyMember={firstName: "Sarah", lastName: "Smith", birthday: "1980-09-03"}


describe('FamilyMember', () => {
  it('renders the FamilyMember component', () => {
    renderWithRouter(<FamilyMember familyMember={familyMember} />)
    
    screen.debug(); // prints out the jsx in the FamilyMember component unto the command line
  })

it("matches snapshot", function() {
  const { asFragment } = renderWithRouter(<FamilyMember familyMember={familyMember} />);
  expect(asFragment()).toMatchSnapshot();
});

})
