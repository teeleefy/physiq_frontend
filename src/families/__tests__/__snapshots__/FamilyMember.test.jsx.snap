// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`FamilyMember > matches snapshot 1`] = `
<DocumentFragment>
  <li
    class="list-group-item"
  >
    <div
      class="FamilyMember-card card"
    >
      <div
        class="card-body"
      >
        <p
          class="card-text"
        />
        <p>
          <b>
            Name: 
          </b>
          <a
            class="FamilyMember-name"
            href="/member/undefined"
          >
            Sarah Smith
          </a>
        </p>
        <p>
          <b>
            Birthday:
          </b>
           1980-09-03
        </p>
        <p />
      </div>
    </div>
  </li>
</DocumentFragment>
`;
